src/node_file.cc

```diff
@@ -1051,6 +1054,34 @@ static void InternalModuleStat(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(rc);
 }
 
+static int32_t FastInternalModuleStat(
+    Local<Object> recv,
+    const FastOneByteString& input,
+    // NOLINTNEXTLINE(runtime/references) This is V8 api.
+    FastApiCallbackOptions& options) {
+  Environment* env = Environment::GetCurrent(recv->GetCreationContextChecked());
+
+  std::string_view path(input.data, input.length);
+  if (UNLIKELY(!env->permission()->is_granted(
+          permission::PermissionScope::kFileSystemRead, path))) {
+    options.fallback = true;
+    return -1;
+  }
```

Comment:
Just thinking about this, if this is true, it will:

- ignore the value -1.
- re-run `InternalModuleStat` but in the slower version.
- throw error since the permission is not granted.

I don't know if worthy but maybe one test just to ensure this behavior, maybe a loop and then pass a value that is not allowed.

Reply:
I added a test (6 months later)